src/app/GitUI/HelperDialogs/FormResetAnotherBranch.cs

```diff
@@ -105,6 +108,11 @@ private void Ok_Click(object sender, EventArgs e)
             bool success = FormProcess.ShowDialog(this, UICommands, arguments: command, Module.WorkingDir, input: null, useDialogSettings: true);
             if (success)
             {
+                if (cbxCheckoutBranch.Checked)
+                {
+                    UICommands.StartCheckoutBranch(this, gitRefToReset.Name);
```

Comment:
> If I don't want to checkout - I'd untick the checkbox.

Because the checkbox state is persisted, the users would need to remember that they have to set that checkbox accordingly _before_ confirming the "Reset other branch" dialog. _I will_ surely forget to do this too often.

I want be asked all the time whether I want to checkout the branch or not.
This could be done by passing an according argument to `StartCheckoutBranch`.

> This appears to be the existing behaviour.
> Personally, I wouldn't want to change it. If there's nothing pending, I wouldn't want a dialog to be shown for me to dismiss (by the virtue of selecting OK).

Sounds like we need a three-state option:
- always checkout automatically
- prompt for checkout
- do not automatically checkout

Reply:
I think you may be overthinking it :) IMO, this dialog falls into "advanced use" category. So, I feel we're discussing a rather nuanced feature.
How often (compared to other dialogs) do you actually use this dialog? I proposed the feature because I have some weird work patterns lately - I often move "other" branch, and I need to switch to it.

If the checkbox isn't ticked - there is no change to the existing behaviours.
If you want to check out "another" branch once it's been moved, then you have the checkbox ticked. I totally get why you'd want to see the checkout dialog if there are pending changes (reset/merge/etc.), but why do you need the checkout dialog if there are no pending changes?

Reply:
I use this a lot, too.

> but why do you need the checkout dialog if there are no pending changes?

Most of the time, I want to confirm the checkout dialog in order to check out the moved branch.
But when I am just tracking the progress of other branches or keeping a copy before rebasing a branch f.i., I do not want to check them out by mistake.
In the current state of this PR,
- I need to untick the checkbox _before_ confirming the reset dialog.
  If I forget this (and I know I will), I must find and checkout my initial branch again and have unnecessarily touched files in my working dir, which can cause lengthy rebuilds. (C++ compilers can take much time.)
- Next time, I need to remember / realize that I need to tick the checkbox again.

Reply:
The checkbox is like checking out a branch, the behavior is what I expect.
I will miss to check/uncheck, hopefully less than I forget to checkout after moving now (normally I see it before pushing, not always).
A second popup can be added, maybe instead of the checkbox, but I prefer as it is. But not a strong opinion.